connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockBlockPickRequestPacketTranslator.java

```diff
@@ -0,0 +1,94 @@
+ *
+ */
+
+package org.geysermc.connector.network.translators.bedrock;
+
+import com.github.steveice10.mc.protocol.data.game.world.block.BlockState;
+import com.github.steveice10.mc.protocol.packet.ingame.client.window.ClientMoveItemToHotbarPacket;
+import com.nukkitx.math.vector.Vector3i;
+import com.nukkitx.protocol.bedrock.packet.BlockPickRequestPacket;
+import com.nukkitx.protocol.bedrock.packet.PlayerHotbarPacket;
+import org.geysermc.connector.entity.Entity;
+import org.geysermc.connector.inventory.Inventory;
+import org.geysermc.connector.network.session.GeyserSession;
+import org.geysermc.connector.network.translators.PacketTranslator;
+import org.geysermc.connector.network.translators.Translator;
+import org.geysermc.connector.network.translators.item.ItemEntry;
+import org.geysermc.connector.network.translators.world.block.BlockTranslator;
+import org.geysermc.connector.network.translators.item.ItemTranslator;
+
+@Translator(packet = BlockPickRequestPacket.class)
+public class BedrockBlockPickRequestPacketTranslator extends PacketTranslator<BlockPickRequestPacket> {
+
+    @Override
+    public void translate(BlockPickRequestPacket packet, GeyserSession session) {
+        Vector3i vector = packet.getBlockPosition();
+        BlockState blockToPick = session.getConnector().getWorldManager().getBlockAt(session, vector.getX(), vector.getY(), vector.getZ());
+
+        // Check if block is null for safety
+        if (blockToPick == null) return;
+        
+        // Block is air - chunk caching is probably off
+        if (blockToPick.getId() == 0) return;
+
+        // Get the inventory to choose a slot to pick
+        Inventory inventory = session.getInventoryCache().getOpenInventory();
+        if (inventory == null) inventory = session.getInventory();
+
+        String targetIdentifier = BlockTranslator.getJavaIdBlockMap().inverse().get(blockToPick).split("\\[")[0];
+        ItemTranslator itemTranslator = new ItemTranslator();
+
+        // Check hotbar for item
+        for (int i = 36; i < 45; i++) {
+            if (inventory.getItem(i) != null) {
+                ItemEntry item = itemTranslator.getItem(inventory.getItem(i));
+                if (item.getJavaIdentifier().equals(targetIdentifier)) {
+                    PlayerHotbarPacket hotbarPacket = new PlayerHotbarPacket();
+                    hotbarPacket.setContainerId(0);
+                    // Java inventory slot to hotbar slot ID
+                    hotbarPacket.setSelectedHotbarSlot(i - 36);
+                    hotbarPacket.setSelectHotbarSlot(true);
+                    session.sendUpstreamPacket(hotbarPacket);
+                    session.getInventory().setHeldItemSlot(i - 36);
+                    // Don't check inventory if item was in hotbar
+                    return;
+                }
+            }
+        }
+
+        // Check inventory for item
+        for (int i = 9; i < 36; i++) {
+            if (inventory.getItem(i) != null) {

```

Comment:
Nesting needs to be cleaned up here as well.

Reply:
Cleaned up?